src/lib/core/exception.h

```diff
@@ -231,14 +292,42 @@ class CryptoError: public Exception {
 	}
 
 	virtual void raise() { throw this; }
+
+	virtual struct error *dup() const
+	{
+		return dup_error<CryptoError>(this);
+	}
 };
 
 class RaftError: public Exception {
 public:
+	RaftError() : Exception(&type_RaftError, NULL, 0) {}
+
 	RaftError(const char *file, unsigned line, const char *format, ...);
 	virtual void raise() { throw this; }
+	virtual struct error *dup() const
+	{
+		return dup_error<RaftError>(this);
+	}
 };
 
+template <class ErrorT>
+struct error *
+Exception::dup_error(const struct error *to_dup)
+{
+	try {
+		struct error *e = new ErrorT;
+		memcpy(e, to_dup, sizeof(*e));
```

Comment:
I'm using `memcpy` with `struct error` which is a C structure. It seems to be an undefined behavior in case of memcpy with `Exception` class because of it's vtabe. I don't understand why in my case this behavior is undefined, could you please explain me?

Reply:
Fixed. Now I'm copying field to field.